torchtitan/utils.py

```diff
@@ -145,7 +148,13 @@ def get_peak_flops(device_name: str) -> int:
             return 1979e12
         elif "PCIe" in device_name:
             return 756e12
-        else:  # for SXM and other variants
+        # Sometimes, the device name is just literally "NVIDIA H100", so
+        # we won't hit the two conditions before; we then need
+        # to check the memory size of the device to determine the peak flops.
+        # H100 NVL has 94 GiB memory, source: https://www.nvidia.com/en-us/data-center/h100/
+        elif device_mem >= 94:
```

Comment:
I don't think this is reliable. There are H100 SXM variants with 96GB. Meta's H100s also have the SXM form factor despite having 96GB HBM2e.

I wonder if we should use `lspci` to determine the form factor.